### PR TITLE
align: init at 1.1.3

### PR DESCRIPTION
**Pull request by hrhino**
###### Motivation for this change

This is a nice package for viewing CSV files and such, where differing field widths can make it hard to understand the data.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/text/align/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{ lib, buildGoPackage, fetchFromGitHub }:
+
+let
+  pname = "align";
+  version = "1.1.3";
+  rev = "v${version}";
+  owner = "Guitarbum722";
+in buildGoPackage rec {
+  inherit pname version rev;
```
Comment (figsoda) [inline pkgs/tools/text/align/default.nix:9]:
```suggestion
buildGoModule rec {
  pname = "align";
  version = "1.1.3";
```

File: pkgs/tools/text/align/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{ lib, buildGoPackage, fetchFromGitHub }:
```
Comment (figsoda) [inline pkgs/tools/text/align/default.nix:1]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub }:
```

File: pkgs/tools/text/align/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+let
+  pname = "align";
+  version = "1.1.3";
+  rev = "v${version}";
+  owner = "Guitarbum722";
+in buildGoPackage rec {
+  inherit pname version rev;
+
+  goPackagePath = "github.com/${owner}/${pname}";
+
```
Comment (figsoda) [inline pkgs/tools/text/align/default.nix:12]:
```suggestion
  vendorSha256 = null;

```

File: pkgs/tools/text/align/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  src = fetchFromGitHub {
+    inherit owner rev;
+    repo = pname;
```
Comment (figsoda) [inline pkgs/tools/text/align/default.nix:15]:
```suggestion
    owner = "Guitarbum722";
    repo = pname;
    rev = "v${version}";
```

File: pkgs/tools/text/align/default.nix [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    sha256 = "17gs3417633z71kc6l5zqg4b3rjhpn2v8qs8rnfrk4nbwzz4nrq3";
+  };
+
+  doCheck = true;
+
```
Comment (figsoda) [inline pkgs/tools/text/align/default.nix:20]:
```suggestion
```

**Review by figsoda — CHANGES_REQUESTED** [review]
[buildGoModule](https://nixos.org/manual/nixpkgs/unstable/#ssec-language-go) is recommended over `buildGoPackage`

File: pkgs/tools/text/align/default.nix (hunk 5 above)
Comment (hrhino) [reply pkgs/tools/text/align/default.nix:20]: why?

**Comment by hrhino** [comment]
good to know about `buildGoModule`; thanks!

File: pkgs/tools/text/align/default.nix (hunk 5 above)
Comment (figsoda) [reply pkgs/tools/text/align/default.nix:17]: `buildGoModule` defaults to `doCheck = true`
https://github.com/NixOS/nixpkgs/blob/3360a376cb905ac40e77d5424089d4893e9cd157/pkgs/development/go-modules/generic/default.nix#L236

Comment (hrhino) [reply pkgs/tools/text/align/default.nix:17]: ah! good to know... I know it's disabled on most builds. Yay go.

File: pkgs/tools/text/align/default.nix [hunk 6]
```diff
@@ -0,0 +1,22 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "align";
+  version = "1.1.3";
+
+  vendorSha256 = null;
```
Comment (figsoda) [inline pkgs/tools/text/align/default.nix:7]: this should go below src

**Review by figsoda — CHANGES_REQUESTED** [review]
otherwise lgtm

File: pkgs/tools/text/align/default.nix (hunk 6 above)
Comment (hrhino) [reply pkgs/tools/text/align/default.nix]: No sooner said than done :)

**Comment by figsoda** [comment]
Result of `nixpkgs-review pr 137893` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>align</li>
  </ul>
</details>

**Review by figsoda — APPROVED** [review]

**Comment by hrhino** [comment]
Thank you!

**Pull request merged**